test/Solver/range1.ll

```diff
@@ -0,0 +1,16 @@
+; REQUIRES: solver
+
+; RUN: %llvm-as -o %t %s
+; RUN: %souper %solver -check %t 2> %t2 || true

```

Comment:
what is the purpose of the || true in this and other tests?

Reply:
If I run this test separately by running all the commands individually, it passes. However, if I put it a test suite without "|| true", it fails. I saw more test files where for the same kind of tests which perform an unexpected simplification, true is added to makes sure that it passes. I did the same here. Honestly, I am not sure why it is done for those tests too. But, I did run it with both debug/release build to find if there is any error in the test, but there is nothing like that because all these tests run successfully when ran individually. Reference: souper/test/Tool/unexpected1.ll

Reply:
some of the existing tests do this to describe the situation where souper fails (return non-zero to the OS) but the test case passes. but your test cases aren't of that kind, so you don't want this in your tests.

Reply:
at least that's what I think is going on, you should make sure.

Reply:
Well, I see the exact same behavior here. I get an exit code 1 but the test case passes. So, I added "true", is that fine?

Reply:
For reference: 
******************** TEST 'Souper :: Solver/range1.ll' FAILED ********************
Script:
--
: 'RUN: at line 3';   /uusoc/exports/scratch/jubi/my_repo/superopt_project/souper/third_party/llvm/Release/bin/llvm-as -o /uusoc/exports/scratch/jubi/my_repo/superopt_project/souper/build/test/Solver/Output/range1.ll.tmp /uusoc/exports/scratch/jubi/my_repo/superopt_project/souper/test/Solver/range1.ll
: 'RUN: at line 4';   /uusoc/exports/scratch/jubi/my_repo/superopt_project/souper/build/souper -z3-path=/uusoc/exports/scratch/jubi/my_repo/tool-installations/install/z3/bin/z3 -check /uusoc/exports/scratch/jubi/my_repo/superopt_project/souper/build/test/Solver/Output/range1.ll.tmp 2> /uusoc/exports/scratch/jubi/my_repo/superopt_project/souper/build/test/Solver/Output/range1.ll.tmp2
: 'RUN: at line 5';   /uusoc/exports/scratch/jubi/my_repo/superopt_project/souper/third_party/llvm/Release/bin/FileCheck /uusoc/exports/scratch/jubi/my_repo/superopt_project/souper/test/Solver/range1.ll < /uusoc/exports/scratch/jubi/my_repo/superopt_project/souper/build/test/Solver/Output/range1.ll.tmp2
--
Exit Code: 1

Reply:
it doesn't seem right
it's souper returning 1, or filecheck? shouldn't souper-check return 0 if the optimization checks out?

Reply:
souper returns 1 here.

Reply:
and what is souper's output?